### ISSUE TITLE
Diff check callback failed (Internal error)

### ISSUE THREAD

**Issue by frimpler** [sysrepo/sysrepo#2332]
Hi Michal,

since sr_session_get_nc_id() has been replaced by sr_session_get_event_nc_id() in some places, we got the rpc-error as mentioned in the subject when trying to edit-config something. The reason seems to be, that here: https://github.com/sysrepo/sysrepo/blob/088752f61ae6ba9a9b62332cedbf2b348c2178e8/src/sysrepo.c#L1959
something like this:
`memcpy(&tmp_sess.ev_sid, &session->ev_sid, sizeof(sr_sid_t));`
is missing.  (at least I fixed it that way)
In other words
np2srv_diff_check_cb() will thow an error, because  np_get_nc_sess_user() won't find the user.
(The root cause is in sysrepo, but the issue is visible in netopeer2.

Regards,
Frank

**Comment by jktjkt** [comment]
Yup, I'm also getting this when I send `<edit-config>`:
```
netopeer2-server[457]: Internal error (/home/jkt/work/prog/_build/_br-cfb/build/netopeer2-custom/src/main.c:511)
netopeer2-server[457]: Failed to send an RPC (User callback failed).
```

**Comment by michalvasko** [comment]
Sorry about that, I have forgot to update this special event case. After updating *sysrepo* and *netopeer2* it should work.

**Comment by jktjkt** [comment]
I wonder what [our test suite](https://github.com/CESNET/netconf-cli/blob/master/tests/datastore_access.cpp), our CI builds and our developer workflow are doing differently compared to "real deployments" on devices. The test suite has definitely sent out `<edit-config>` around. @syyyr suggested that it could be due to the "real deployment" running as root; is this related, perhaps? Note that in all cases the connection still goes via the UNIX socket.

**Comment by jktjkt** [comment]
OK, changing the test suite to set `SYSREPO_SUPERUSER_UID` and `NACM_RECOVERY_UID` reproduces this problem. I'll extend our test coverage I guess.

**Comment by frimpler** [comment]
Thanks, Michal. Seems to work.
